podcast/2.7.md

```diff
@@ -0,0 +1,717 @@
+
+**Beyang Liu:**
+So the SDK is a thing that you embed in the client. And then the server side is something that's running elsewhere and the SDK will talk to the server to determine which flags are on and how should I affect the behavior of the application.
+
+**Ivar Østhus:**
+Yes, it will synchronize with the server in the background and also that part is very optimized. That's different per SDK, of course, per platform, in the Java SDK we will actually only use a single trend in the background to do all the communication with the Unleash API just to minimize our resource consumption on your application. And it's all those small details that are super important to be appealing to developers. You also have the APIs of course, where you can do anything you can do in the management UI, you can do also from the API. This is also something we have a lot of plans on improving, making it even easier to do patches and stuff like that. So that you can update just what you actually want to update and not send longer payloads. So we have a lot of plans on even improving the APIs to make it even simpler.
+
+**Ivar Østhus:**
+But you also really interact with the management UI to define your feature toggle... Actually, you don't have to actually manually define it because if you start using a feature toggle in the SDK it will actually detect it and it will prompt you to actually create that feature toggle in the management UI, if you haven't already.
+
+**Beyang Liu:**
+Oh, that's cool.
+
+**Ivar Østhus:**
+Because we will send usage metrics back to the API so that we know which apps are connected, how many instances are they connected from, and which feature toggles do they actually consume?
+
+**Beyang Liu:**
+Interesting.
+
+**Ivar Østhus:**
+So we will collect all that user data, and from that, we can make it easier to understand, is this feature toggle actually being used? Where is it being used? How often is it used in the last hour? Stuff like that. Because the other part of this is the technical depth side of things. This is also a huge problem with feature toggles. So you are creating a way, the long running feature branches that are never merged back to master because it's hard. And then you have the feature toggles instead, which actually are feature branches inside your code, giving you a lot of power. But then you have all these different code paths in your application. And for some reason, it seems like people don't clean up afterwards. A lot of developers are moving on, moving their focuses to new features and it seems like it's nice to just leave those feature toggles around. This bugs me a lot and I think we, as a tool provider, should also address this issue because the amount of feature toggles is not the point in itself.
```

Comment:
```suggestion
So we will collect all that user data, and from that, we can make it easier to understand, is this feature toggle actually being used? Where is it being used? How often is it used in the last hour? Stuff like that. Because the other part of this is the technical debt side of things. This is also a huge problem with feature toggles. So you are creating a way, the long running feature branches that are never merged back to master because it's hard. And then you have the feature toggles instead, which actually are feature branches inside your code, giving you a lot of power. But then you have all these different code paths in your application. And for some reason, it seems like people don't clean up afterwards. A lot of developers are moving on, moving their focuses to new features and it seems like it's nice to just leave those feature toggles around. This bugs me a lot and I think we, as a tool provider, should also address this issue because the amount of feature toggles is not the point in itself.
```

Reply:
"Tech debt" seems to the phrase/concept that makes the most sense in this context (and probably the prior).